README.md

```diff
@@ -1,2 +1,174 @@
 # qrcode.js
-A QrCode library
+![license](https://img.shields.io/github/license/mashape/apistatus.svg)
+![circleCi](https://circleci.com/gh/AdactiveSAS/qrcode.js.svg?style=shield&circle-token=:circle-token)
+
+<p align="center">
+  <a href="http://adactive.com">
+    <img alt="babel" src="https://user-images.githubusercontent.com/8574893/41667613-f9636b62-74df-11e8-9b9a-981f86a2a521.png">
+  </a>
+</p>
+
+## Introduction
+qrCode.js is a web based QrCode generation library in javascript. It extends the functionality of [qrcode-generator](https://github.com/kazuhikoarase/qrcode-generator). 
+
+## Getting started
+
+```javascript
+    npm i qrcode.es
+```
+OR
+```javascript
+    yarn add qrcode.es
+```
+
+## Usage
+
+### ES6 Modules
+```javascript
+import QrCode, {MODES} from 'qrcode.es';
+    
+const qrCodeSetting = {
+    size: 400,
+    minVersion: 8,
+    text: 'http://adactive.com/',
+    background: '#fff',
+    mode: MODES.DRAW_WITH_IMAGE_BOX,
+    radius: 0.5,
+    image: 'https://raw.githubusercontent.com/AdactiveSAS/qrcode.js/master/adactiveLogo.jpg',
+    mSize: 0.15,
+};
+
+const ele = document.getElementById("qrCode"); //Element must be an instance of HTMLCanvasElement or HTMLDivElement
+const qrCode = new QrCode(ele); //Initializing the QrCode
+await qrCode.generate(qrCodeSetting); // Function that generates the QrCode
+let image = this.qrCode.getImage(); // Function to get the data Url of the QrCode Image
+
+```
+
+### Browser
+```html
+<!DOCTYPE html>
+<html>
+    <head>
+        <style>
+            html, body {
+                height: 100%;
+                width: 100%;
+                position: relative;
+                margin: 0;
+                overflow: hidden;
+            }
+        </style>
+    </head>
+    <body>
+        <div id="qrCode"></div>
+
+        <!-- Assuming node_modules is located in parent directory -->
+        <script src="../node_modules/qrcode.umd.js"></script>
+        <script type="application/javascript">
+
+            const image = './assets/logo.jpg'; // To modify path to the image url
+            const ele = document.getElementById("qrCode"); //Element must be an instance of HTMLCanvasElement or HTMLDivElement
+            const qrCode = new QrCode.qrcode(ele);
+
+            const qrCodeSetting = {
+                size: 400,
+                minVersion: 8,
+                text: 'http://adactive.com/',
+                background: '#fff',
+                mode: QrCode.MODES.DRAW_WITH_IMAGE_STRIP,
+                radius: 0.5,
+                image: image,
+                mSize: 0.15,
+            };
+
+            let qrCodeImageUrl = null;
+
+            qrCode.generate(qrCodeSetting)
+                .then(() => {
+                    qrCodeImageUrl = qrCode.getImage();
+                    console.log("[IMAGE URL] : ", qrCodeImageUrl);
+                });
+
+        </script>
+    </body>
+</html>
+
+```
+
+## Functions
```

Comment:
Should be API section and the generate signature is not true and miss the constructor